src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_UnaryOperator.cs

```diff
@@ -1028,29 +1041,45 @@ private static BinaryOperatorKind GetCorrespondingBinaryOperator(BoundIncrementO
             return result;
         }
 
-        private static ConstantValue GetConstantOneForBinOp(
+        private static (TypeSymbol, ConstantValue) GetConstantOneForIncrement(
+            CSharpCompilation compilation,
             BinaryOperatorKind binaryOperatorKind)
         {
+            ConstantValue constantOne;
             switch (binaryOperatorKind.OperandTypes())
             {
                 case BinaryOperatorKind.PointerAndInt:
                 case BinaryOperatorKind.Int:
-                    return ConstantValue.Create(1);
+                    constantOne = ConstantValue.Create(1);
+                    break;
                 case BinaryOperatorKind.UInt:
-                    return ConstantValue.Create(1U);
+                    constantOne = ConstantValue.Create(1U);
+                    break;
                 case BinaryOperatorKind.Long:
-                    return ConstantValue.Create(1L);
+                    constantOne = ConstantValue.Create(1L);
+                    break;
                 case BinaryOperatorKind.ULong:
-                    return ConstantValue.Create(1LU);
+                    constantOne = ConstantValue.Create(1LU);
+                    break;
+                case BinaryOperatorKind.NInt:
+                    constantOne = ConstantValue.Create(1);
+                    return (compilation.GetSpecialType(SpecialType.System_IntPtr).AsNativeInt(true), constantOne);
+                case BinaryOperatorKind.NUInt:
+                    constantOne = ConstantValue.Create(1U);
+                    return (compilation.GetSpecialType(SpecialType.System_UIntPtr).AsNativeInt(true), constantOne);
```

Comment:
>AsNativeInt [](start = 83, length = 11)

nit: there is a big of ambiguity when we say "native int". It can be `nint` or `nint` + `nuint`. So this method's name is a bit ambiguous. I'm not sure I have a much better idea though...

Reply:
Yes, the caller is responsible for calling this method on the appropriate underlying type.

---
In reply to: [379042816](https://github.com/dotnet/roslyn/pull/41380#discussion_r379042816) [](ancestors = 379042816)